334 docs rfc create the docs for tigerlake (#4194)
timescale/docs: use-timescale/tigerlake/index.md

---
title: TigerLake

> excerpt: Unifies the Timescale Cloud operational architecture with the data lake (S3 + Iceberg) architectures. This enables real-time application building alongside efficient data pipeline management within a single system.

Reviewer: ```suggestion
excerpt: Unifies the Tiger Cloud operational architecture with the data lake (S3 + Iceberg) architectures. This enables real-time application building alongside efficient data pipeline management within a single system
```

products: [cloud]
keywords: [data lake]
---

# TigerLake

> TigerLake Unifies the Timescale Cloud operational architecture with the data lake (S3 + Iceberg) architectures. This enables real-time application building alongside efficient data pipeline management within a single system.

Reviewer: ```suggestion
TigerLake unifies the $CLOUD_LONG operational architecture with the data lake (S3 + Iceberg) architectures. This enables real-time application building alongside efficient data pipeline management within a single system.
```

This experimental release is a native integration enabling continuous replication
between AWS S3 Tables (managed Iceberg and catalog) running in your AWS account and

> relational tables and hypertables in Timescale Cloud. 

Reviewer: ```suggestion
relational tables and hypertables in $CLOUD_LONG. 
```

> You interact directly with TigerLake using Timescale Console. The pricing associated

Reviewer: ```suggestion
You interact directly with TigerLake using $CONSOLE_LONG. The pricing associated
```

with the integration will be introduced at a later date. This document explains how to

> get started for the early access partners. There are no costs associated with the Timescale Cloud service, we set everything up and then invite you to the project.

Reviewer: ```suggestion
get started for the early access partners. There are no costs associated with the $SERVICE_LONG—we set everything up and then invite you to the project.
```

## Get started

> You first create an S3 table bucket, then use an with an ARN role to enable Timescale Cloud to write to this bucket. Timescale Console supplies a CloudFormation template. You setup TigerCLoud

Reviewer: then use an ARN role?

> You first create an S3 table bucket, then use an with an ARN role to enable Timescale Cloud to write to this bucket. Timescale Console supplies a CloudFormation template. You setup TigerCLoud

Reviewer: ```suggestion
You first create an S3 table bucket, then use an with an ARN role to enable $CLOUD_LONG to write to this bucket. $CONSOLE_LONG supplies a CloudFormation template. You set up $CLOUD_LONG
```

using either:

- AWS Management Console

> - AWS cloudformation CLI

Reviewer: ```suggestion
- AWS CloudFormation CLI
```

> ### Setup TigerLake using AWS Management Console

Reviewer: ```suggestion
### Set up TigerLake using AWS Management Console
```

1. Sign in to the AWS Management Console and open  [CloudFormation console][cmc].
1. In the navigation bar on the top of the page:

>    1. Choose the name of the currently displayed AWS Region

Reviewer: I can see that Amazon capitalizes Region, but i don't think we do it anywhere in our docs.

>    1. Choose the name of the currently displayed AWS Region

Reviewer: ```suggestion
   1. Choose the name of the currently displayed AWS Region.
```

   2. Set it to the Region in which you want to create your table bucket.
   

>    This must match the region your Timescale service is running in. If the regions do

Reviewer: ```suggestion
   This must match the region your $SERVICE_LONG is running in. If the regions do
```

>    not match AWS charges you for cross-region data transfer.

Reviewer: ```suggestion
   not match, AWS charges you for cross-region data transfer.
```

1. Click `Create stack`. 

>    If prompted choose `With new resources`. This is the standard option.

Reviewer: ```suggestion
   If prompted, choose `With new resources`. This is the standard option.
```

> 1. Under `Specify Template`, copy the following URL into the Amazon S3 URL box and Click `Next`.

Reviewer: ```suggestion
1. Under `Specify Template`, copy the following URL into the Amazon S3 URL box and click `Next`.
```

>    ```https://tigerlake.s3.us-east-1.amazonaws.com/tigerlake-connect-cloudformation.yaml```

Reviewer: ```suggestion
   ```yaml
   https://tigerlake.s3.us-east-1.amazonaws.com/tigerlake-connect-cloudformation.yaml
   ```
```

1. Enter the following details, then click `Next`:
   - **Stack Name**: the name for this CloudFormation stack

>    - **BucketName**: The name of the S3 table bucket which will be created

Reviewer: ```suggestion
   - **BucketName**: the name of the S3 table bucket which will be created
```

Reviewer: Technically I think those need to be in the code font, not bold?

Reviewer 2: Yup, let me address that separately

>    - **ProjectID and ServiceID**: Your Timescale Cloud service details (see these instructions)

Reviewer: ```suggestion
   - **ProjectID and ServiceID**: Your $SERVICE_LONG details (see these instructions)
```

1. Check `I acknowledge that AWS CloudFormation might create IAM resources`, then click `Next`.
1. On the review page, click `Submit` and wait for the deployment to complete. 
1. Click `Outputs`, then copy all four outputs. 

> 1. Provide the outputs to Timescale.  

Reviewer: ```suggestion
1. Provide the outputs to $COMPANY.  
```

>    Timescale uses them to provision your TigerLake services.

Reviewer: ```suggestion
  $COMPANY uses them to provision your $SERVICE_LONGs.
```

> ### Setup TigerLake using the aws cloudformation CLI

Reviewer: ```suggestion
### Set up TigerLake using the AWS CloudFormation CLI
```

Copy the following command, and replace the values for:
- **Stack Name**: the name for this CloudFormation stack

> - **BucketName**: The name of the S3 table bucket which will be created

Reviewer: ```suggestion
- **BucketName**: the name of the S3 table bucket which will be created
```

> - **ProjectID and ServiceID**: Your Timescale Cloud service details (see these instructions)

Reviewer: ```suggestion
- **ProjectID and ServiceID**: your $SERVICE_LONG details (see these instructions)
```

Reviewer: Same comment about code font vs. bold

```shell
aws cloudformation create-stack \
--capabilities CAPABILITY_IAM \
--template-url https://tigerlake.s3.us-east-1.amazonaws.com/tigerlake-connect-cloudformation.yaml \
--stack-name {STACK_NAME} \
--parameters \
ParameterKey=BucketName,ParameterValue="{BUCKET_NAME}" \
ParameterKey=ProjectID,ParameterValue="{ProjectID}" \
ParameterKey=ServiceID,ParameterValue="{ServiceID}"
```

## Start streaming

> To stream a table from a Timescale Cloud service to Iceberg, run the following statement:

Reviewer: ```suggestion
To stream a table from a $SERVICE_LONG to Iceberg, run the following statement:
```

```sql
SELECT create_iceberg_sync('<TABLE_NAME>'::regclass);
```

When a stream is started, the full table is synchronized to Iceberg, meaning that all prior records will be imported first.

## Query your data

To execute queries against Iceberg, best practice is to use the following:

- [AWS Athena][aws-athena]: ensure that integration with AWS analytics services is enabled for the table bucket.
- [duckdb][duckdb]: support for S3 Tables is in preview. 
- [Apache Spark][apache-spark]

## Gotchas and known issues

> - Only Postgres 17 is supported.

Reviewer: ```suggestion
- Only $PG 17 is supported.
```

- The only supported Iceberg catalog is the S3 Tables REST catalog.
- It is not possible to stop a stream from TimescaleDB to Iceberg at the moment, this ability is under implementation.
- When streaming a hypertable to Iceberg, the Iceberg table will be partitioned with a hardcoded one-day partition interval.
- Only tables with primary keys are supported.

> - We have some limitations on the changes you can make to a table after setting up Iceberg sync : dropping/renaming columns and modifying a column’s data type is not permitted. We will add support for this soon.

Reviewer: ```suggestion
- We have some limitations on the changes you can make to a table after setting up Iceberg sync: dropping/renaming columns and modifying a column’s data type is not permitted. We will add support for this soon.
```

> - Certain optimizations must be disabled in Hypertables with the columnstore enabled to retrieve correlating WAL events.

Reviewer: ```suggestion
- Certain optimizations must be disabled in hypertables with the columnstore enabled to retrieve correlating WAL events.
```
